validator.py

```diff
@@ -942,73 +1050,36 @@ def verify_allowed_actions(self):
         '''
         # inital variables for getting specific data 
         data = copy.deepcopy(self.loaded_yaml)
-        supplies = data['state']['supplies']
         scenes = data['scenes']
 
-        # determine the first scene
-        first_scene = self.determine_first_scene(data)
-
-        # checking whether the pulse oximeter exists in the inital state 
-        first_scene_po = False   
-        for item in supplies:
-            if item['type'] == 'Pulse Oximeter' and item['quantity'] > 0:
-                first_scene_po = True
-
-        # checking whether the first defined scene is valid based on the pulse oximeter in the inital state
-        first_map = first_scene['action_mapping']   
-        if not first_scene_po:
-            for action in first_map:
-                if action['action_type'] == 'CHECK_BLOOD_OXYGEN' or action['action_type'] == 'CHECK_ALL_VITALS':
-                    self.pulse_ox_info(first_scene['id'], True)
-                    break
-
-        # testing all other scenes if they are valid in terms of their actions and the pulse oximeter   
         for scene in scenes:
-            if scene is not first_scene:
-                if 'state' in scene:
-                    scene_state = scene['state']
-
-                    # if supplies are not defined in the state and there is no pulse oximeter originally available
-                    if 'supplies' not in scene_state and not first_scene_po:
-                        # check if invalid action type exists when the pulse oximeter is not available
-                        map = scene['action_mapping']
-                        for action in map:
-                            if action['action_type'] == 'CHECK_BLOOD_OXYGEN' or action['action_type'] == 'CHECK_ALL_VITALS':
-                                self.pulse_ox_info(scene['id'], False)
-                                break
-                            
-                    # if supplies are defined in the scene
-                    if 'supplies' in scene_state:
-                        pulse_ox = False
-                        for item in scene_state['supplies']:
-                            if item['type'] == 'Pulse Oximeter' and item['quantity'] > 0:
-                                pulse_ox = True
-                        # check if invalid action type exists when the pulse oximeter is not available
-                        map = scene['action_mapping']
-                        for action in map:
-                            if action['action_type'] == 'CHECK_BLOOD_OXYGEN' or action['action_type'] == 'CHECK_ALL_VITALS':
-                                if not pulse_ox:
-                                    self.pulse_ox_info(scene['id'], True)
-                                    break
-                                    
+            for action in scene.get('action_mapping', []):
+                if action['action_type'] in ['CHECK_BLOOD_OXYGEN', 'CHECK_ALL_VITALS']:
```

Comment:
I'd do a `continue` here to reduce the levels of nesting

```
if action['action_type'] not in ['CHECK_BLOOD_OXYGEN', 'CHECK_ALL_VITALS']: 
    continue
```

Reply:
Updated